src/app/PubSubVitessceGrid.js

```diff
@@ -3,11 +3,50 @@ import VitessceGrid from 'vitessce-grid';
 
 import { LayerPublisher } from '../components/layerpublisher';
 
+function getFromLS(key) {
+  let ls;
+  if (window.localStorage) {
+    try {
+      ls = JSON.parse(window.localStorage.getItem(key)) || {};
+    } catch (e) {
+      console.error(`Error accessing ${key} in local storage.`);
+    }
+  }
+  return ls;
+}
+
 export default class PubSubVitessceGrid extends React.Component {
   constructor(props) {
     super(props);
+    const storageSavedLayouts = getFromLS('vitGrid');
     this.rowHeight = props.rowHeight;
+    this.config = props.config;
+    this.getStorageLayout = () => {
+      const dataSetSavedLayout = storageSavedLayouts[props.config.name] || [];
+      // update with user's saved configs
+      for (let i = 0; i < dataSetSavedLayout.length; i += 1) {
```

Comment:
Fix logic to avoid assuming config layout data length === save layout data length